src/agent/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.12.4)
+
+project(WazuhAgent)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_BUILD_TYPE Release)
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g")
+
+include_directories(${CMAKE_SOURCE_DIR})
+
+# Add subdirectories
+add_subdirectory(communicator)
```

Comment:
We could commit this line and the files added into the `communicator` folder on a separate commit, so you have one commit adding the cmake changes to build the agent together with the hello world in `agent.cpp` on one, and the `communicator` related code on another.